### ISSUE TITLE
Non release tags are protected from deletion.

### ISSUE THREAD

**Issue by robwdwd** [go-gitea/gitea#2644]
- Gitea version (or commit ref): 1.2.0+rc3-1-g4c9bf91
- Git version: 2.11.0
- Operating system: Ubuntu 16.10 
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

When creating a local tag and pushing it to gitea it creates that tag as a release but if I want to then delete the remote tag it rejects even though this tag has not been used as a release when creating a release from the web UI. I can also not delete the tag/release from the web UI as you can do if you create a release within the web UI itself.
```

$ git tag test3
$ git push origin test3
Total 0 (delta 0), reused 0 (delta 0)
To server:user/test.git
 * [new tag]         test3 -> test3

$ git push --delete origin test3
remote: Gitea: branch refs/tags/test3 is protected from deletion
To server:user/test.git
 ! [remote rejected] test3 (pre-receive hook declined)
error: failed to push some refs to 'git@server:user/test.git'
```

**Comment by daviian** [comment]
Please test with latest changes on release/v1.2 branch. Should be solved with #2630

**Comment by lafriks** [comment]
Should be fixed with #2630. Please reopen if you still have issues
